### DISCUSSION TITLE
How to enable tracing for custom created classes without code change.

### DISCUSSION THREAD

**Discussion by mahudees** [apache/skywalking#10765] [General]
Would like to have the application logic code classes also to be instrumented and get trace from it. Is there any way to get that without any code change, like giving a class name in config should add instrumentation for all the methods in the class.

**Comment by wu-sheng** [comment]
This is the doc, https://skywalking.apache.org/docs/skywalking-java/v8.15.0/en/setup/service-agent/java-agent/customize-enhance-trace/

As you are adding instrument codes by yourself, you need to make sure  spans created correctly.

**Reply by mahudees** [reply]
Thank you very much.
